provider/consumer.py

```diff
@@ -485,7 +485,7 @@ def __encodeMessageIfNeeded(self, value):
 
         if self.encodeValueAsJSON:
             try:
-                parsed = json.loads(value)
+                parsed = json.loads(value, parse_constant=self.__errorOnJSONConstant)
```

Comment:
do the json.dumps calls in this module also need to specify `allow_nan=false` (default is `true`)?

Reply:
The `json.dumps` call needs to be able to read JSON constants to get an accurate message size value.

Reply:
ah.. i see it now
```
    def __sizeMessage(self, message):
        messagePayload = self.__getMessagePayload(message)
        return len(json.dumps(messagePayload).encode('utf-8'))
```
the json is validated in the `__getMessagePayload()` call that occurs before the `json.dumps` call.